decisions/DR006-Generic-string-type.md

```diff
@@ -0,0 +1,179 @@
+  ([VFX reference platform](https://vfxplatform.com/) is currently stuck
+  on the old ABI).
+* [ACCU Overload, 21(116):11-15, 2013](https://accu.org/journals/overload/21/116/steinbach_1842/)
+  article on portable string literals.
+* [utf8everywhere](http://utf8everywhere.org/) article promoting "just
+  use UTF-8".
+* Google Abseil library's [string types/functions](https://abseil.io/docs/cpp/guides/strings)
+  has some notes on `string_view` constants and lifetime.
+* vfx-rs's [cppmm](https://github.com/vfx-rs/cppmm) semi-automated
+  C++->C  bindings generator.
+* [UTF8-CPP](https://github.com/nemtrif/utfcpp) C++ library provides
+  functions for operating on UTF-8 strings stored within a
+  `std::string`.
+* IBM's [ICU](https://icu.unicode.org/) popular heavyweight C (unicode)
+  string library with custom types.
+* [CsString](https://github.com/copperspice/cs_string) C++ library
+  providing dedicated unicode-aware string types.
+* [LabText](https://github.com/meshula/LabText) C-native string library
+  with C++ wrappers.
+* [SDS](https://github.com/antirez/sds) Redis's C-native string library.
+
+
+## Options considered
+
+### Option 1
+
+`std::string_view`/`std::string` for constants/parameters.
+
+#### Pros
+
+- Little boilerplate required.
+- Straightforward manipulation by `std::string`-compatible library
+  functions.
+- Consistent with other VFX open source projects (see above).
+
+#### Cons
+
+- Ties us to a specific underlying class that cannot be extended.
+- Sensitive to ABI differences between compilers (e.g. GCC C++11 ABI
+  break, see above).
+- Sensitive to cross-platform implementation differences.
+
+Estimated cost: Small
+
+### Option 2
+
+Custom type with hidden implementation ([pimpl](https://en.cppreference.com/w/cpp/language/pimpl)).
+
+#### Pros
+
+- Compiler/library isolation.
+- Allows use of intrusive smart pointers (e.g.
+  `enable_shared_from_this`).
+- Can be extended with custom logic for validation, encoding, etc.
+- Implementation can be extended without breaking ABI compatibility of
+  hosts/plugins.
+- A third-party library can be used under-the-hood.
+
+#### Cons
+
+- Additional boilerplate required, e.g. converting to/from native types,
+  and custom Python pybind11 converters.
+- (Small) performance hit of additional pointer indirection to
+  implementation.
+
+Estimated cost: Medium
+
+### Option 3
+
+Exposing a third-party library.
+
+#### Pros
+
+- No reinventing the wheel for (common) string functions.
+- Consistent implementation across compilers.
+- A C-native library (with C++ wrappers) would avoid the need for roll
+  our own C bindings.
+
+#### Cons
+
+- Additional third-party dependency - especially problematic if exposed
+  as a core type required by adopters, and even worse if adopters are
+  already using a different (or customized) version of the library.
+- Custom Python converters required.
+
+Estimated cost: Medium
+
+## Outcome
+
+We've tentatively decided on Option 1.
```

Comment:
[fix] Based on the discussion, I'd say remove the tentative for now, and I'll take the responsibility longer term if we have any issues.

Reply:
Removed in f2485c7a